src/parseWithPointers.ts

```diff
@@ -12,30 +22,408 @@ export const parseWithPointers: SourceMapParser = <T>(value: string): IParserRes
 
   if (!value || !value.trim().length) return parsed;
 
-  const result = jsonSourceMap.parse(value);
-  parsed.data = result.data;
-  parsed.pointers = transformPointers(result.pointers);
+  const validations: IValidationResult[] = [];
 
-  return parsed;
+  return {
+    ...parse(value, validations, {
+      disallowComments: true,
+    }),
+    validations,
+  };
 };
 
-const transformPointers = (pointers: any) => {
-  const transformed: IParserResultPointers = {};
+// based on jsonc-parser source code
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+interface IJSONVisitor extends JSONVisitor {
+  onNewLine?: (lineNumber: ParseErrorCode, offset: number, length: number) => void;
+}
 
-  for (const path in pointers) {
-    if (!pointers.hasOwnProperty(path)) continue;
+function toPointer(...fragments: Segment[]) {
```

Comment:
I had a look in the whole source code and apparently this is possible, so we should be able to change this.

Reply:
why though? how does this improve code quality?

Reply:
This function is currently accepting comma separated arguments, and they all get accumulated in a single array so that I can use it in this way:

`toPointer(a, b, c, d, e)`

Thing is — such commodity is never used and there's instead a conversion happening everywhere to have it in such format:

https://github.com/stoplightio/json/pull/10/files#diff-69de2470cb4743d24ba9bf89b77916d3R76
https://github.com/stoplightio/json/pull/10/files#diff-69de2470cb4743d24ba9bf89b77916d3R88
https://github.com/stoplightio/json/pull/10/files#diff-69de2470cb4743d24ba9bf89b77916d3R105

…basically we're doing a conversion for a byproduct we're not leveraging.

Does that make sense? @P0lip

Reply:
We leverage it once, at line 76